### ISSUE TITLE
Improve error reporting in GenerateCompilerInternals.targets

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/roslyn#7267]
I've seen spurious/intermittent build errors from CSharpSyntaxGenerator returning exit code 9009 (invalid path).

It would be nice to add <Error Condition="!(Exists('$(File)'))" Text="..." /> tasks to all targets in 
https://github.com/dotnet/roslyn/blob/master/build/Targets/GenerateCompilerInternals.targets

that verify that all files that it needs to work indeed exist on disk and paths are not malformed. That way next time it fails we'll get a nice error message ("file ... not found") instead of just the 9009 exit code.

**Comment by davkean** [comment]
Where/how? Can you get me a build log next time this happens?

**Comment by KirillOsenkov** [comment]
As far as I could tell Build.cmd doesn't result in a log file on disk. And since you've trimmed down the verbosity to make builds faster due to console slowness bug, we have no way of investigating. I recommend threading the /flp:v=diag argument somewhere to the actual MSBuild invocation so that it drops a log on disk.

**Comment by jaredpar** [comment]
Remove the flaky label as there are no repros for us to investigate.

**Comment by KirillOsenkov** [comment]
I think this was fixed recently with the build ordering fix (BasicScripting project referencing some other one, I can't remember). @jaredpar I think you fixed it.

**Comment by KirillOsenkov** [comment]
oh! I remember now. This was prompted by issue https://github.com/dotnet/roslyn/issues/7869 (which is now fixed). However the GenerateCompilerInternals.targets are written in such a way that if any of the inputs are not found, it fails silently with an error code 9009. 

This issue is about adding additional <Error> statements to GenerateCompilerInternals.targets that would output a better error message in case any of the inputs are missing in the future. This would make it significantly easier to understand what file was missing (instead of just mysterious exit code 9009).

I remember I spent a lot of time trying to figure out https://github.com/dotnet/roslyn/issues/7869, and if we had better error logging in here, it would be much easier to understand.
